### PR TITLE
[R4R] Add flags to auction queries

### PR DESCRIPTION
**Pull request by denalimarsh**
Adds flags `type`, `denom`, and `phase` to cli and rest for auction queries.

To test:
1. `kvcli tx cdp create 100000000bnb 10000000usdx --from testuser --gas 250000`
2. `kvcli tx pricefeed postprice bnb:usd 1.001 99999999999 --from validator`
3. `kvcli q auction auctions --type=Collateral --denom=bnb --phase=Forward`

### CODE REVIEW

File: x/auction/client/cli/query.go [hunk 1]
```diff
@@ -68,30 +78,97 @@ func QueryGetAuctionCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
...
 func QueryGetAuctionsCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
-	return &cobra.Command{
+	cmd := &cobra.Command{
 		Use:   "auctions",
-		Short: "get a list of active auctions",
-		Args:  cobra.NoArgs,
+		Short: "query auctions with optional filters",
+		Long: strings.TrimSpace(`Query for all paginated auctions that match optional filters:
+Example:
+$ kvcli q auction auctions --type=(Collateral|Surplus|Debt)
```
Comment (fedekunze) [inline x/auction/client/cli/query.go:86]:
```suggestion
$ kvcli q auction auctions --type=(collateral|surplus|debt)
```

File: x/auction/client/cli/query.go [hunk 2]
```diff
@@ -68,30 +78,97 @@ func QueryGetAuctionCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
...
+$ kvcli q auction auctions --denom=bnb
+$ kvcli q auction auctions --phase=(Forward|Reverse)
```
Comment (fedekunze) [inline x/auction/client/cli/query.go:88]:
```suggestion
$ kvcli q auction auctions --phase=(forward|reverse)
```

File: x/auction/client/cli/query.go [hunk 3]
```diff
@@ -68,30 +78,97 @@ func QueryGetAuctionCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
...
+				return fmt.Errorf("No matching auctions found")
+			}
 
 			auctionsWithPhase := []types.AuctionWithPhase{} // using empty slice so json returns [] instead of null when there's no auctions
-			for _, a := range auctions {
+			for _, a := range matchingAuctions {
 				auctionsWithPhase = append(auctionsWithPhase, types.NewAuctionWithPhase(a))
 			}
-			return cliCtx.PrintOutput(auctionsWithPhase)
+			cliCtx = cliCtx.WithHeight(height)
```
Comment (fedekunze) [inline x/auction/client/cli/query.go:160]: we should add this too for the other auction queries

File: x/auction/client/rest/query.go [hunk 4]
```diff
@@ -68,22 +70,70 @@ func queryAuctionHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
...
 		cliCtx, ok := rest.ParseQueryHeightOrReturnBadRequest(w, cliCtx, r)
 		if !ok {
 			return
 		}
 
-		// Get all auctions
-		res, height, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", types.ModuleName, types.QueryGetAuctions), nil)
+		var auctionType string
+		var auctionDenom string
+		var auctionPhase string
```
Comment (fedekunze) [inline x/auction/client/rest/query.go:87]: nit, group vars: 

```
var (
...
)
```

File: x/auction/keeper/querier_test.go [hunk 5]
```diff
@@ -107,7 +107,9 @@ func (suite *QuerierTestSuite) TestQueryAuctions() {
 	// Set up request query
 	query := abci.RequestQuery{
 		Path: strings.Join([]string{custom, types.QuerierRoute, types.QueryGetAuctions}, "/"),
-		Data: types.ModuleCdc.MustMarshalJSON(types.NewQueryAllAuctionParams(1, TestAuctionCount)),
+		Data: types.ModuleCdc.MustMarshalJSON(
```
Comment (karzak) [inline x/auction/keeper/querier_test.go:110]: Add issue to update tests to include tests with flags - low priority.

Comment (denalimarsh) [reply x/auction/keeper/querier_test.go:110]: https://github.com/Kava-Labs/kava/issues/526

File: x/auction/client/rest/query.go (hunk 4 above)
Comment (denalimarsh) [reply x/auction/client/rest/query.go:87]: https://github.com/Kava-Labs/kava/pull/522/commits/94c1ebd775c02c8f01c1ad32745b29d4ba58d4f5

File: x/auction/client/cli/query.go (hunk 3 above)
Comment (denalimarsh) [reply x/auction/client/cli/query.go:164]: https://github.com/Kava-Labs/kava/pull/522/commits/94c1ebd775c02c8f01c1ad32745b29d4ba58d4f5

File: x/auction/client/cli/query.go (hunk 2 above)
Comment (denalimarsh) [reply x/auction/client/cli/query.go]: https://github.com/Kava-Labs/kava/pull/522/commits/94c1ebd775c02c8f01c1ad32745b29d4ba58d4f5

File: x/auction/client/cli/query.go (hunk 1 above)
Comment (denalimarsh) [reply x/auction/client/cli/query.go]: https://github.com/Kava-Labs/kava/pull/522/commits/94c1ebd775c02c8f01c1ad32745b29d4ba58d4f5

**Pull request merged**